### PR TITLE
feat: CEC in M365 Copilot: implement citation in rag ts&js template

### PR DESCRIPTION
**Pull request by QinghuiMeng-M**
implement citation in rag ts&js template
feature: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/29180531/

### CODE REVIEW

File: templates/js/custom-copilot-rag-azure-ai-search/src/app/customSayCommand.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+Object.defineProperty(exports, "__esModule", { value: true });
```
Comment (kuojianlu) [inline templates/js/custom-copilot-rag-azure-ai-search/src/app/customSayCommand.js:1]: Any reason to use this way?
module.exports = { xxx } not workable?

File: templates/js/custom-copilot-rag-azure-ai-search/src/app/customSayCommand.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+exports.sayCommand = void 0;
+const botbuilder = require("botbuilder");
+const Utilities = require("@microsoft/teams-ai");
+/**
+ * @private
+ * @param {boolean} feedbackLoopEnabled - If true, the feedback loop UI for Teams will be enabled.
+ * @returns {''} - An empty string.
+ */
```
Comment (kuojianlu) [inline templates/js/custom-copilot-rag-azure-ai-search/src/app/customSayCommand.js:9]: Remove the comment

Comment (QinghuiMeng-M) [reply templates/js/custom-copilot-rag-azure-ai-search/src/app/customSayCommand.js]: ok,removed

File: templates/js/custom-copilot-rag-azure-ai-search/src/app/customSayCommand.js (hunk 1 above)
Comment (QinghuiMeng-M) [reply templates/js/custom-copilot-rag-azure-ai-search/src/app/customSayCommand.js]: Here is just copy from SayCommand.js
looks like it is not required, so I remove it and code still work

File: templates/js/custom-copilot-rag-azure-ai-search/src/app/customSayCommand.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+          "@id": "",
+          additionalType: ["AIGeneratedContent"],
+          ...(referencedCitations ? { citation: referencedCitations } : {}),
+        },
+      ],
+    });
+    return "";
+  };
+}
+exports.sayCommand = sayCommand;
```
Comment (kuojianlu) [inline templates/js/custom-copilot-rag-azure-ai-search/src/app/customSayCommand.js:65]: Update to module.exports = { ... }, to be consistent with azureAISearchDataSource.js

Comment (QinghuiMeng-M) [reply templates/js/custom-copilot-rag-azure-ai-search/src/app/customSayCommand.js]: ok, updated

File: templates/js/custom-copilot-rag-azure-ai-search/src/app/customSayCommand.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
+const botbuilder = require("botbuilder");
+const Utilities = require("@microsoft/teams-ai");
+
+function sayCommand(feedbackLoopEnabled = false) {
+  return async (context, _state, data) => {
+    if (!data.response?.content) {
+      return "";
+    }
+    let content = "";
+    let result = JSON.parse(data.response.content);
```
Comment (kuojianlu) [inline templates/js/custom-copilot-rag-azure-ai-search/src/app/customSayCommand.js:10]: Add error handling, e.g., if not valid json, just send as raw text

Comment (QinghuiMeng-M) [reply templates/js/custom-copilot-rag-azure-ai-search/src/app/customSayCommand.js]: ok, added

File: templates/js/custom-copilot-rag-azure-ai-search/src/app/customSayCommand.js [hunk 5]
```diff
@@ -6,15 +6,35 @@ function sayCommand(feedbackLoopEnabled = false) {
...
     }
-    let content = "";
-    let result = JSON.parse(data.response.content);
     const isTeamsChannel = context.activity.channelId === botbuilder.Channels.Msteams;
+    let content = "";
+    let result = undefined;
+    try {
+      result = JSON.parse(data.response.content);
+    } catch (error) {
+      console.error(`Response unformat, send the raw text. error: ${error}`);
```
Comment (kuojianlu) [inline templates/js/custom-copilot-rag-azure-ai-search/src/app/customSayCommand.js:15]: Response is not valid json, ...

Comment (QinghuiMeng-M) [reply templates/js/custom-copilot-rag-azure-ai-search/src/app/customSayCommand.js]: sure

**Review by kuojianlu — APPROVED** [review]

**Pull request merged**